app/components/app_search_results_component.rb

```diff
@@ -19,10 +21,11 @@ class AppSearchResultsComponent < ViewComponent::Base
     <% end %>
   ERB
 
-  def initialize(pagy, label:, heading: "Search results")
+  def initialize(pagy, label:, heading: "Search results", hint_text: nil)
```

Comment:
```suggestion
  def initialize(pagy, label:, heading: "Search results", no_results_text: nil)
```

I think this clarifies where the label is shown (as `hint_text` isn't shown when `has_results?` is `true`).